src/main/java/com/habicus/core/controller/v1/goal/GoalController.java

```diff
@@ -61,7 +72,101 @@ public void setUserService(UserService userService) {
   }
 
   /**
-   * Allows retrieval of user goals based on sec. token
+   * Enable deletion of a particular {@link Goal} record by a requesting user
+   *
+   * @param goalId
+   * @param principal
+   * @return
+   */
+  @DeleteMapping("/goal/{goalId}")
+  public ResponseEntity<ImmutableMap<String, String>> createUserGoal(
+      @PathVariable("goalId") int goalId, Principal principal) {
+    Goal deletedGoal;
+    ImmutableMap<String, String> responseData;
+    try {
+      deletedGoal = goalService.removeGoal(principal, goalId);
+      responseData =
+          ControllerUtil.createResponse(
+              "Deleted Goal Successfully!", deletedGoal.toString(), HttpStatus.OK);
+    } catch (Exception e) {
+      responseData =
+          ControllerUtil.createResponse(
+              "Failure to remove specified goal.", HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    return new ResponseEntity<>(responseData, HttpStatus.CREATED);
```

Comment:
May want a different response.. if it ever got here, getting a "CREATED" returned might be weird for the delete method